Hudl.Mjolnir/Util/ConcurrentDictionaryExtensions.cs

```diff
@@ -12,5 +12,11 @@ internal static class ConcurrentDictionaryExtensions
             var lazy = dictionary.GetOrAdd(key, new Lazy<V>(() => valueFactory(key), LazyThreadSafetyMode.PublicationOnly));
```

Comment:
Maybe call the method below? Also I think we spoke about `PublicationOnly` before, should we make the default `ExecutionAndPublication`, since the most common use case is to not allow the `valueFactory` to be called more than once.